File: frontend/src/app/typedefs.ts

export interface ITickerAction {
    ticker: string
    action: "search"|"process"|"report"
}

export interface IResponseData {
    Ticker: string
    Datasource: string
    Payload: any
}

export interface INotification {
    level: "success"|"error"|"warning"|"info"
    message: string
    read: boolean
    datetime: Date
}

File: cmd/router/typedefs.go
package main

type TickerActionType string

var (
	TickerActionTypeSearch  TickerActionType = "search"
	TickerActionTypeProcess TickerActionType = "process"
	TickerActionTypeReport  TickerActionType = "report"
)

type ITickerAction struct {
	Ticker string           `json:"ticker"`
	Action TickerActionType `json:"action"`
}
